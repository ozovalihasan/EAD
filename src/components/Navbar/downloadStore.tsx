import { State } from "@/zustandStore";

const downloadStore = (data: State, projectName: string) => {
  const filename = `${projectName || "EAD"}.json`;
  const { version, idCounter, nodes, edges, tables, orderedTables } = data;
  const jsonData = { version, idCounter, nodes, edges, tables, orderedTables };
  
  const jsonString = JSON.stringify(jsonData, null, 4);
  
  const blob = new Blob([jsonString], { type: 'application/json' });
  const event = document.createEvent('MouseEvents');
  const anchor = document.createElement('a');
  
  anchor.download = filename;
  anchor.href = window.URL.createObjectURL(blob);
  anchor.dataset.downloadurl = ['text/json', anchor.download, anchor.href].join(':');
  event.initMouseEvent('click', true, false, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);
  anchor.dispatchEvent(event);
};

export default downloadStore;
