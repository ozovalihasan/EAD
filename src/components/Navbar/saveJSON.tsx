const saveJSON = (data: object, filename= 'EAD.json') => {
  
    let dataNew: object | string = data;
    const filenameNew = filename;
  
    if (typeof dataNew === 'object') {
      dataNew = JSON.stringify(dataNew, undefined, 4);
    }
  
    const blob = new Blob([dataNew], { type: 'text/json' });
    const e = document.createEvent('MouseEvents');
    const a = document.createElement('a');
  
    a.download = filenameNew;
    a.href = window.URL.createObjectURL(blob);
    a.dataset.downloadurl = ['text/json', a.download, a.href].join(':');
    e.initMouseEvent('click', true, false, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);
    a.dispatchEvent(e);
  };
  
  export default saveJSON;