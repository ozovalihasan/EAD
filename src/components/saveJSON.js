/* eslint-disable no-console */
const saveJSON = (data, filename) => {
  let dataNew = data;
  let filenameNew = filename;
  if (!dataNew) {
    console.error('No data');
    return;
  }

  if (!filenameNew) { filenameNew = 'console.json'; }

  if (typeof dataNew === 'object') {
    dataNew = JSON.stringify(dataNew, undefined, 4);
  }

  const blob = new Blob([dataNew], { type: 'text/json' });
  const e = document.createEvent('MouseEvents');
  const a = document.createElement('a');

  a.download = filenameNew;
  a.href = window.URL.createObjectURL(blob);
  a.dataset.downloadurl = ['text/json', a.download, a.href].join(':');
  e.initMouseEvent('click', true, false, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);
  a.dispatchEvent(e);
};

export default saveJSON;
