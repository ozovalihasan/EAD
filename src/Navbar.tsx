import AngleDown from './AngleDown';
import saveJSON from './components/saveJSON';
import EADLogo from './EADLogo';
import GithubLogo from './GithubLogo';
import UpArrow from './UpArrow';
import useCustomizationStore from './zustand/customizationStore';
import useStore from './zustand/store';

const Navbar = () => {

  const { 
    resetStore,
    uploadStore,
    
  } = useStore();

  const { 
    navbarVisible,
    toggleNavbarVisibility
   } = useCustomizationStore()
  
  return (
    <div className='relative'>
      <nav className={`bg-transparent relative h-20 py-2 border-b border-solid border-first-500 ${navbarVisible || 'hidden'}`} >
        <div className='flex items-center gap-4' >
          <div className="flex items-center mr-8">
            <a className='h-16 w-16 mr-2 ml-8' href="#">
              <EADLogo />
            </a>
            <div className='text-first-500 text-end text-xs '>
              0.4.0
            </div>
          </div>
          <button
              className="rounded-md p-3 my-1 ml-0 btn-first"
              onClick={() => saveJSON(useStore.getState(), 'EAD.json')}
              type="button"
              title="Download EAD"
            >
              Download EAD
          </button>
      
          <label className='text-center p-3 rounded-md cursor-pointer btn-second' title="Upload an EAD file">
            Upload EAD
            <input
              className='hidden'
              onChange={uploadStore}
              type="file"
              accept=".json"
              title="Select an uploaded EAD"
            />
          </label>
      
          <button
            className="rounded-md p-3 my-1 mx-0 btn-second "
            onClick={resetStore}
            type="button"
            title="Reset"
          >
            Reset
          </button>
      
          <a className='w-8 h-8 mx-8 ml-auto' href="https://github.com/ozovalihasan/ead" target="_blank" title="The repository of the project">
            <GithubLogo />
          </a>
      
        </div>
      
      </nav>
      <div 
        className="z-50 block top-0 flex w-full h-0 border  hover:cursor-move hover:bg-first-300 justify-center" 
        title="Click to hide/show the navbar." 
        
      >
        <div className="h-3 w-3 z-50" onClick={toggleNavbarVisibility} >
          {
            navbarVisible ? <UpArrow /> : <AngleDown />
          }
        </div>
      </div>
    </div>
  )
}

export default Navbar;