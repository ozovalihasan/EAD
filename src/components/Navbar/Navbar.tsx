import saveJSON from './saveJSON';
import {
  EADLogo,
  GithubLogo,
  Settings
} from "@/components";

import {
  UpArrow, 
  AngleDown
} from "@/icons";
import { useCustomizationStore } from '@/zustandStore/customizationStore';
import { useStore } from '@/zustandStore';

export const Navbar = () => {

  
  const resetStore = useStore(store => store.resetStore)
  const uploadStore = useStore(store => store.uploadStore)

  const navbarVisible = useCustomizationStore(store => store.navbarVisible)
  const toggleNavbarVisibility = useCustomizationStore(store => store.toggleNavbarVisibility)


  return (
    <div className='relative'>
      <nav className={`bg-transparent relative h-20 py-2 border-b general-border ${navbarVisible ? "" : 'hidden'}`} >
        <div className='flex items-center gap-4' >
          <div className="flex items-center mr-8">
            <a className='h-16 w-16 mr-2 ml-8' href="#">
              <EADLogo />
            </a>
            <div className='text-first-500 text-end text-xs '>
              0.4.7
            </div>
          </div>
          <button
              className="rounded-md p-3 my-1 ml-0 btn-first"
              onClick={() => saveJSON(useStore.getState())}
              type="button"
              title="Download EAD"
            >
              Download EAD
          </button>
      
          <label className='text-center p-3 rounded-md cursor-pointer btn-second' title="Upload an EAD file">
            Upload EAD
            <input
              className='hidden'
              onChange={uploadStore}
              type="file"
              accept=".json"
              title="Select an uploaded EAD"
            />
          </label>
      
          <button
            className="rounded-md p-3 my-1 mx-0 btn-second "
            onClick={resetStore}
            type="button"
            title="Reset"
          >
            Reset
          </button>

          <div className='ml-auto'>
            <Settings />
          </div>
      
          <a className='w-8 h-8 mx-8 ml-8' href="https://github.com/ozovalihasan/ead" target="_blank" title="The repository of the project">
            <GithubLogo />
          </a>
      
        </div>
      
      </nav>
      <button 
        title="Click to show/hide the navbar" 
        className="absolute left-0 -bottom-8 h-6 w-6 z-50 m-2 btn-third rounded-full" 
        onClick={toggleNavbarVisibility} 
      >
        <div className="stroke-[40] w-3 h-3">
          {
            navbarVisible ? <UpArrow /> : <AngleDown />
          }
        </div>
      </button>
    </div>
  )
}