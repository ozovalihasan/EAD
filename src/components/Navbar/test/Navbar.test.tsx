import {  Navbar } from '../Navbar';
import { render, screen, renderHook, fireEvent } from "@testing-library/react";
import useStore from '@/zustandStore/store';
import useCustomizationStore from '@/zustandStore/customizationStore';
import saveJSON from "../saveJSON"

jest.mock('../saveJSON',  () => ({
  default: jest.fn()
}))

jest.mock('@/components',  () => ({
  EADLogo: (
    () => (
      <>
        MockEADLogo
      </>
    )
  ),
  GithubLogo: (
    ( ) => (
      <>
        MockGithubLogo
      </>
    )
  ),
}))
  

let renderReadyComponent: JSX.Element;

beforeEach(() => {
    
  useStore.setState({ 
    resetStore: jest.fn(),
    uploadStore: jest.fn()
  })
    
  renderReadyComponent = (
    <Navbar />
  );
});

describe('<Navbar />', () => {
  
  describe('if navbar is visible', () => {


    beforeEach(() => {
      useCustomizationStore.setState({ 
        navbarVisible: true
      })
    });

    it('renders EADLogo component', () => {

      render(renderReadyComponent );

      expect(screen.getByText(/MockEADLogo/i)).toBeInTheDocument();
    });

    it('renders a button to download a EAD', () => {

      render(renderReadyComponent );

      const downloadButton = screen.getByRole("button", { name: "Download EAD" });
      expect(downloadButton).toBeInTheDocument();

      fireEvent.click(downloadButton);

      expect(saveJSON).toHaveBeenCalledTimes(1);;
    });
    
    it('renders a button to upload a EAD', () => {

      render(renderReadyComponent );

      const { result } = renderHook(() => useStore());

      const uploadButton = screen.getByText("Upload EAD");

      expect(uploadButton).toBeInTheDocument();

      fireEvent.change(uploadButton.getElementsByTagName('input')[0], "");
      
      expect(result.current.uploadStore).toHaveBeenCalledTimes(1);;
    });

    it('renders a button to download a EAD', () => {

      render(renderReadyComponent );

      const { result } = renderHook(() => useStore());

      const resetButton = screen.getByRole("button", { name: "Reset" });
      expect(resetButton).toBeInTheDocument();

      fireEvent.click(resetButton);

      expect(result.current.resetStore).toHaveBeenCalledTimes(1);;
    });

    it('renders GithubLogo component as an anchor', () => {

      render(renderReadyComponent );

      const githubElement = screen.getByText(/MockGithubLogo/i)

      expect(githubElement).toBeInTheDocument();
      expect(githubElement.tagName).toEqual("A");
    });

    it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent );
      expect(renderedContainer).toMatchSnapshot();
    });
  })  

  describe('if navbar is invisible', () => {


    beforeEach(() => {
      useCustomizationStore.setState({ 
        navbarVisible: false
      })
    });

    it('has a "hidden" class', () => {
      render(renderReadyComponent );

      const navElement = screen.getAllByText("")[0].getElementsByTagName('nav')[0]

      expect(navElement.classList).toContain("hidden");
    });


    it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent );
      expect(renderedContainer).toMatchSnapshot();
    });
  })

});