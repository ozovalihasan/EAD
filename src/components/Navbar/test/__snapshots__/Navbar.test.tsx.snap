// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> if navbar is invisible renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative"
      >
        <nav
          class="bg-transparent relative h-20 py-2 border-b general-border hidden"
        >
          <div
            class="flex items-center gap-4"
          >
            <div
              class="flex items-center mr-8"
            >
              <a
                class="h-16 w-16 mr-2 ml-8"
                href="#"
              >
                MockEADLogo
              </a>
              <div
                class="text-first-500 text-end text-xs "
              >
                0.4.7
              </div>
            </div>
            <button
              class="rounded-md p-3 my-1 ml-0 btn-first"
              title="Download EAD"
              type="button"
            >
              Download EAD
            </button>
            <label
              class="text-center p-3 rounded-md cursor-pointer btn-second"
              title="Upload an EAD file"
            >
              Upload EAD
              <input
                accept=".json"
                class="hidden"
                title="Select an uploaded EAD"
                type="file"
              />
            </label>
            <button
              class="rounded-md p-3 my-1 mx-0 btn-second "
              title="Reset"
              type="button"
            >
              Reset
            </button>
            <div
              class="ml-auto"
            >
              MockSettings
            </div>
            <a
              class="w-8 h-8 mx-8 ml-8"
              href="https://github.com/ozovalihasan/ead"
              target="_blank"
              title="The repository of the project"
            >
              MockGithubLogo
            </a>
          </div>
        </nav>
        <button
          class="absolute left-0 -bottom-8 h-6 w-6 z-50 m-2 btn-third rounded-full"
          title="Click to show/hide the navbar"
        >
          <div
            class="stroke-[40] w-3 h-3"
          >
            MockAngleDown
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative"
    >
      <nav
        class="bg-transparent relative h-20 py-2 border-b general-border hidden"
      >
        <div
          class="flex items-center gap-4"
        >
          <div
            class="flex items-center mr-8"
          >
            <a
              class="h-16 w-16 mr-2 ml-8"
              href="#"
            >
              MockEADLogo
            </a>
            <div
              class="text-first-500 text-end text-xs "
            >
              0.4.7
            </div>
          </div>
          <button
            class="rounded-md p-3 my-1 ml-0 btn-first"
            title="Download EAD"
            type="button"
          >
            Download EAD
          </button>
          <label
            class="text-center p-3 rounded-md cursor-pointer btn-second"
            title="Upload an EAD file"
          >
            Upload EAD
            <input
              accept=".json"
              class="hidden"
              title="Select an uploaded EAD"
              type="file"
            />
          </label>
          <button
            class="rounded-md p-3 my-1 mx-0 btn-second "
            title="Reset"
            type="button"
          >
            Reset
          </button>
          <div
            class="ml-auto"
          >
            MockSettings
          </div>
          <a
            class="w-8 h-8 mx-8 ml-8"
            href="https://github.com/ozovalihasan/ead"
            target="_blank"
            title="The repository of the project"
          >
            MockGithubLogo
          </a>
        </div>
      </nav>
      <button
        class="absolute left-0 -bottom-8 h-6 w-6 z-50 m-2 btn-third rounded-full"
        title="Click to show/hide the navbar"
      >
        <div
          class="stroke-[40] w-3 h-3"
        >
          MockAngleDown
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Navbar /> if navbar is visible renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative"
      >
        <nav
          class="bg-transparent relative h-20 py-2 border-b general-border "
        >
          <div
            class="flex items-center gap-4"
          >
            <div
              class="flex items-center mr-8"
            >
              <a
                class="h-16 w-16 mr-2 ml-8"
                href="#"
              >
                MockEADLogo
              </a>
              <div
                class="text-first-500 text-end text-xs "
              >
                0.4.7
              </div>
            </div>
            <button
              class="rounded-md p-3 my-1 ml-0 btn-first"
              title="Download EAD"
              type="button"
            >
              Download EAD
            </button>
            <label
              class="text-center p-3 rounded-md cursor-pointer btn-second"
              title="Upload an EAD file"
            >
              Upload EAD
              <input
                accept=".json"
                class="hidden"
                title="Select an uploaded EAD"
                type="file"
              />
            </label>
            <button
              class="rounded-md p-3 my-1 mx-0 btn-second "
              title="Reset"
              type="button"
            >
              Reset
            </button>
            <div
              class="ml-auto"
            >
              MockSettings
            </div>
            <a
              class="w-8 h-8 mx-8 ml-8"
              href="https://github.com/ozovalihasan/ead"
              target="_blank"
              title="The repository of the project"
            >
              MockGithubLogo
            </a>
          </div>
        </nav>
        <button
          class="absolute left-0 -bottom-8 h-6 w-6 z-50 m-2 btn-third rounded-full"
          title="Click to show/hide the navbar"
        >
          <div
            class="stroke-[40] w-3 h-3"
          >
            MockUpArrow
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative"
    >
      <nav
        class="bg-transparent relative h-20 py-2 border-b general-border "
      >
        <div
          class="flex items-center gap-4"
        >
          <div
            class="flex items-center mr-8"
          >
            <a
              class="h-16 w-16 mr-2 ml-8"
              href="#"
            >
              MockEADLogo
            </a>
            <div
              class="text-first-500 text-end text-xs "
            >
              0.4.7
            </div>
          </div>
          <button
            class="rounded-md p-3 my-1 ml-0 btn-first"
            title="Download EAD"
            type="button"
          >
            Download EAD
          </button>
          <label
            class="text-center p-3 rounded-md cursor-pointer btn-second"
            title="Upload an EAD file"
          >
            Upload EAD
            <input
              accept=".json"
              class="hidden"
              title="Select an uploaded EAD"
              type="file"
            />
          </label>
          <button
            class="rounded-md p-3 my-1 mx-0 btn-second "
            title="Reset"
            type="button"
          >
            Reset
          </button>
          <div
            class="ml-auto"
          >
            MockSettings
          </div>
          <a
            class="w-8 h-8 mx-8 ml-8"
            href="https://github.com/ozovalihasan/ead"
            target="_blank"
            title="The repository of the project"
          >
            MockGithubLogo
          </a>
        </div>
      </nav>
      <button
        class="absolute left-0 -bottom-8 h-6 w-6 z-50 m-2 btn-third rounded-full"
        title="Click to show/hide the navbar"
      >
        <div
          class="stroke-[40] w-3 h-3"
        >
          MockUpArrow
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Navbar /> it renders a button to show/hide the navbar renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative"
      >
        <nav
          class="bg-transparent relative h-20 py-2 border-b general-border "
        >
          <div
            class="flex items-center gap-4"
          >
            <div
              class="flex items-center mr-8"
            >
              <a
                class="h-16 w-16 mr-2 ml-8"
                href="#"
              >
                MockEADLogo
              </a>
              <div
                class="text-first-500 text-end text-xs "
              >
                0.4.7
              </div>
            </div>
            <button
              class="rounded-md p-3 my-1 ml-0 btn-first"
              title="Download EAD"
              type="button"
            >
              Download EAD
            </button>
            <label
              class="text-center p-3 rounded-md cursor-pointer btn-second"
              title="Upload an EAD file"
            >
              Upload EAD
              <input
                accept=".json"
                class="hidden"
                title="Select an uploaded EAD"
                type="file"
              />
            </label>
            <button
              class="rounded-md p-3 my-1 mx-0 btn-second "
              title="Reset"
              type="button"
            >
              Reset
            </button>
            <div
              class="ml-auto"
            >
              MockSettings
            </div>
            <a
              class="w-8 h-8 mx-8 ml-8"
              href="https://github.com/ozovalihasan/ead"
              target="_blank"
              title="The repository of the project"
            >
              MockGithubLogo
            </a>
          </div>
        </nav>
        <button
          class="absolute left-0 -bottom-8 h-6 w-6 z-50 m-2 btn-third rounded-full"
          title="Click to show/hide the navbar"
        >
          <div
            class="stroke-[40] w-3 h-3"
          >
            MockUpArrow
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative"
    >
      <nav
        class="bg-transparent relative h-20 py-2 border-b general-border "
      >
        <div
          class="flex items-center gap-4"
        >
          <div
            class="flex items-center mr-8"
          >
            <a
              class="h-16 w-16 mr-2 ml-8"
              href="#"
            >
              MockEADLogo
            </a>
            <div
              class="text-first-500 text-end text-xs "
            >
              0.4.7
            </div>
          </div>
          <button
            class="rounded-md p-3 my-1 ml-0 btn-first"
            title="Download EAD"
            type="button"
          >
            Download EAD
          </button>
          <label
            class="text-center p-3 rounded-md cursor-pointer btn-second"
            title="Upload an EAD file"
          >
            Upload EAD
            <input
              accept=".json"
              class="hidden"
              title="Select an uploaded EAD"
              type="file"
            />
          </label>
          <button
            class="rounded-md p-3 my-1 mx-0 btn-second "
            title="Reset"
            type="button"
          >
            Reset
          </button>
          <div
            class="ml-auto"
          >
            MockSettings
          </div>
          <a
            class="w-8 h-8 mx-8 ml-8"
            href="https://github.com/ozovalihasan/ead"
            target="_blank"
            title="The repository of the project"
          >
            MockGithubLogo
          </a>
        </div>
      </nav>
      <button
        class="absolute left-0 -bottom-8 h-6 w-6 z-50 m-2 btn-third rounded-full"
        title="Click to show/hide the navbar"
      >
        <div
          class="stroke-[40] w-3 h-3"
        >
          MockUpArrow
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
